mixer/tools/codegen/pkg/modelgen/model.go

```diff
@@ -417,8 +418,12 @@ func getTypeNameRec(g *FileDescriptorSetParser, field *descriptor.FieldDescripto
 	case descriptor.FieldDescriptorProto_TYPE_BOOL:
 		return TypeInfo{Name: "bool"}, TypeInfo{Name: sBOOL}, nil
 	case descriptor.FieldDescriptorProto_TYPE_MESSAGE:
-		if valueTypeAllowed && field.GetTypeName()[1:] == fullProtoNameOfValueMsg {
-			return TypeInfo{Name: fullProtoNameOfValueTypeEnum, IsValueType: true}, TypeInfo{Name: fullGoNameOfValueTypeEnum, IsValueType: true}, nil
+		if field.GetTypeName()[1:] == fullProtoNameOfValueMsg {
+			if valueTypeAllowed {
+				return TypeInfo{Name: fullProtoNameOfValueTypeEnum, IsValueType: true}, TypeInfo{Name: fullGoNameOfValueTypeEnum, IsValueType: true}, nil
+			} else {
+				return TypeInfo{}, TypeInfo{}, createInvalidTypeError(field.GetName(), valueTypeAllowed, nil)
+			}
```

Comment:
Is there a test covering the behavior change?

Reply:
Yes. https://github.com/istio/istio/blob/master/mixer/tools/codegen/pkg/modelgen/testdata/UnsupportedValueTypeInAPA.proto